lib/sisimai/bite/email/exchange2007.rb

```diff
@@ -0,0 +1,178 @@
+module Sisimai::Bite::Email
+  module Exchange2007
+    # Sisimai::Bite::Email::Exchange2007 parses a bounce email which created by
+    # Microsoft Exchange Server 2007.
+    # Methods in the module are called from only Sisimai::Message.
+    class << self
+      # Imported from p5-Sisimail/lib/Sisimai/Bite/Email/Exchange2007.pm
+      require 'sisimai/bite/email'
+
+      Re0 = {
+        :'subject'          => %r/\AUndeliverable:/,
+        :'content-language' => %r/\A[a-z]{2}(?:[-][A-Z]{2})?\z/,
+      }
+      Re1 = {
+        :begin  => %r/[ ]Microsoft[ ]Exchange[ ]Server[ ]20\d{2}/,
+        :error  => %r/[ ]((?:RESOLVER|QUEUE)[.][A-Za-z]+(?:[.]\w+)?);/,
+        :rhost  => %r/\AGenerating[ ]server:[ ]?(.*)/,
+        :rfc822 => %r/\AOriginal message headers:/,
+        :endof  => %r/\A__END_OF_EMAIL_MESSAGE__\z/,
+      }
+      NDRSubject = {
+        :'SMTPSEND.DNS.NonExistentDomain'=> 'hostunknown',   # 554 5.4.4 SMTPSEND.DNS.NonExistentDomain
+        :'SMTPSEND.DNS.MxLoopback'       => 'networkerror',  # 554 5.4.4 SMTPSEND.DNS.MxLoopback
+        :'RESOLVER.ADR.BadPrimary'       => 'systemerror',   # 550 5.2.0 RESOLVER.ADR.BadPrimary
+        :'RESOLVER.ADR.RecipNotFound'    => 'userunknown',   # 550 5.1.1 RESOLVER.ADR.RecipNotFound
+        :'RESOLVER.ADR.ExRecipNotFound'  => 'userunknown',   # 550 5.1.1 RESOLVER.ADR.ExRecipNotFound
+        :'RESOLVER.ADR.RecipLimit'       => 'toomanyconn',   # 550 5.5.3 RESOLVER.ADR.RecipLimit
+        :'RESOLVER.ADR.InvalidInSmtp'    => 'systemerror',   # 550 5.1.0 RESOLVER.ADR.InvalidInSmtp
+        :'RESOLVER.ADR.Ambiguous'        => 'systemerror',   # 550 5.1.4 RESOLVER.ADR.Ambiguous, 420 4.2.0 RESOLVER.ADR.Ambiguous
+        :'RESOLVER.RST.AuthRequired'     => 'filtered',      # 550 5.7.1 RESOLVER.RST.AuthRequired
+        :'RESOLVER.RST.NotAuthorized'    => 'rejected',      # 550 5.7.1 RESOLVER.RST.NotAuthorized
+        :'RESOLVER.RST.RecipSizeLimit'   => 'mesgtoobig',    # 550 5.2.3 RESOLVER.RST.RecipSizeLimit
+        :'QUEUE.Expired'                 => 'expired',       # 550 4.4.7 QUEUE.Expired
+      }
+      Indicators = Sisimai::Bite::Email.INDICATORS
+
+      def description; return 'Microsoft Exchange Server 2007'; end
+      def smtpagent;   return Sisimai::Bite.smtpagent(self); end
+      def headerlist;  return ['Content-Language']; end
+      def pattern;     return Re0; end
+
+      # Parse bounce messages from Microsoft Exchange Server 2007
+      # @param         [Hash] mhead       Message headers of a bounce email
+      # @options mhead [String] from      From header
+      # @options mhead [String] date      Date header
+      # @options mhead [String] subject   Subject header
```

Comment:
__[rubocop]__
* Style/Semicolon: Do not use semicolons to terminate expressions. [:ref](https://github.com/bbatsov/ruby-style-guide#no-semicolon)
* Layout/TrailingWhitespace: Trailing whitespace detected. [:ref](https://github.com/bbatsov/ruby-style-guide#no-trailing-whitespace)

<!-- Comment by [SideCI](https://sideci.com) -->